sunpy/map/mapbase.py

```diff
@@ -1537,16 +1537,25 @@ def submap(self, bottom_left, *, top_right=None, width: (u.deg, u.pix)=None, hei
         elif (not all([arg is None or (isinstance(arg, u.Quantity) and arg.unit.is_equivalent(u.pix))
                        for arg in (bottom_left, top_right, width, height)])):
             raise TypeError("When bottom_left is not a SkyCoord, any values of top_right, "
-                             "width or height specified must be Quantity objects in units of pixels.")
+                            "width or height specified must be Quantity objects in units of "
+                            "pixels.")
 
         elif bottom_left.shape != (2,) or (top_right is not None and top_right.shape != (2,)):
-            raise ValueError("Both bottom_left and top_right when specified as Quantity objects must have shape (2,)")
+            raise ValueError("Both bottom_left and top_right when specified as Quantity objects "
+                             "must have shape (2,)")
 
         elif height is not None and width is not None:
             top_right = u.Quantity([bottom_left[0] + width, bottom_left[1] + height])
 
         x_pixels = u.Quantity([bottom_left[0], top_right[0]]).to_value(u.pix)
         y_pixels = u.Quantity([top_right[1], bottom_left[1]]).to_value(u.pix)
+        if x_pixels[0] > x_pixels[1]:
+            warnings.warn("The rectangle is inverted in the left/right (longitude) direction,  "
```

Comment:
I don't think adding latitude & longitude to these error messages is a good idea, there is no reason other than convention that the left/right direction is normally longitude. You can easily construct WCSes where this would appear to be right but would actually be wrong and all manner of other weirdness.